lib/charms/magma_orc8r_libs/v0/orc8r_base.py

```diff
@@ -104,27 +110,56 @@ def __init__(
             self.charm.on, f"{service_name_with_underscores}_pebble_ready"
         )
         self.container = self.charm.unit.get_container(self.container_name)
-        self.framework.observe(pebble_ready_event, self._on_magma_orc8r_pebble_ready)
+        self.framework.observe(pebble_ready_event, self._configure_workload)
+        self.framework.observe(self.charm.on.upgrade_charm, self._configure_workload)
 
         if additional_environment_variables:
             self.additional_environment_variables = additional_environment_variables
         else:
             self.additional_environment_variables = {}
```

Comment:
This is not from your PR, but I think this might be better
```suggestion
       self.additional_environment_variables = additional_environment_variables or {}
```

Reply:
I like that. Changed.